src/components/PageDefinitions.js

```diff
@@ -275,12 +304,16 @@ export class PageDefinitions extends AbstractPageDefinitions {
   }
 
   onTextDrop = content => {
-    const contentObject = asObject(content)
-    if (contentObject) return this.onAddComponent(EntitySpec.fromCoordinates(contentObject))
     try {
+      const contentObject = asObject(content)
+      // if the dropped text can be directly interpreted as JSON, assume it's a list
+      if (contentObject) return this.onAddComponent(EntitySpec.fromCoordinates(contentObject))
+      // if it's a URL to a gist, load the gist files
+      if (content.startsWith('https://gist.github.com')) return this.onGistDrop(content)
```

Comment:
I think it's not fair to put this kind of condition here.
We could end up managing different URLs (another one that we already know to manage is PR URL)
In my #356 this have been managed by an high level class Provider, that will address all the different providers, as you can see here https://github.com/clearlydefined/website/pull/356/files#diff-b83a5a3eca3acc3a9eebddb8273448f3R17

What do you think about it?

Reply:
yes. the entire drop thing should be factored out and be pluggable to accept different handlers etc. The providers approach in #356 was too monolithic for this and mixed concerns.